### PR TITLE
Planning: only reuse path when there are no blocking obstacles

### PR DESCRIPTION
**Pull request by sjiang2018**

### CODE REVIEW

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc [hunk 1]
```diff
@@ -39,35 +41,60 @@ PathReuseDecider::PathReuseDecider(const TaskConfig& config)
...
-      !reference_line_info->is_path_lane_borrow()) {
-    ++total_path_counter_;  // count total path
+  // check front static blocking obstacle
+  auto* mutable_path_reuse_decider_status = PlanningContext::Instance()
+                                                ->mutable_planning_status()
+                                                ->mutable_path_reuse_decider();
+  auto* mutable_path_decider_status = PlanningContext::Instance()
+                                          ->mutable_planning_status()
+                                          ->mutable_path_decider();
+  const int kWaitCycle = -2;  // wait 2 cycle
```
Comment (panjiacheng) [inline modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: Use constexpr

File: modules/planning/proto/planning_status.proto [hunk 2]
```diff
@@ -74,6 +74,10 @@ message PathDeciderStatus {
   repeated LaneBorrowDirection decided_side_pass_direction = 5;
 }
 
+message PathReuseDeciderStatus  {
+    optional bool reused_path = 1 [default = true];
```
Comment (xiaoxq) [inline modules/planning/proto/planning_status.proto]: Unify indents, especially in the same file.

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc [hunk 3]
```diff
@@ -39,35 +41,60 @@ PathReuseDecider::PathReuseDecider(const TaskConfig& config)
 
 Status PathReuseDecider::Process(Frame* const frame,
                                  ReferenceLineInfo* const reference_line_info) {
+  if (!Decider::config_.path_reuse_decider_config().reuse_path())
+    return Status::OK();
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: Always add brackets for name scopes.

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc [hunk 4]
```diff
@@ -39,35 +41,60 @@ PathReuseDecider::PathReuseDecider(const TaskConfig& config)
...
-  if (!IsSameStopObstacles(frame, reference_line_info))
-    ADEBUG << "not same stop obstacle";
-  return (IsCollisionFree(reference_line_info) &&
-          IsSameStopObstacles(frame, reference_line_info));
+  //   if (!IsSameStopObstacles(frame, reference_line_info))
+  //     ADEBUG << "not same stop obstacle";
+  //   return (IsCollisionFree(reference_line_info) &&
+  //           IsSameStopObstacles(frame, reference_line_info));
   //   return IsSameStopObstacles(frame, reference_line_info);
   //   return IsSameObstacles(reference_line_info);
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: Either remove unused code, or add `TODO` to state why it's not used, which `return` will be used in what kind of situation. This piece of code is likely to be unmaintainable very soon.

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc [hunk 5]
```diff
@@ -39,35 +41,122 @@ PathReuseDecider::PathReuseDecider(const TaskConfig& config)
...
+  double blocking_obstacle_start_s;
+  if (GetBlockingObstacleS(reference_line_info, &blocking_obstacle_start_s) &&
+      // distance to blocking obstacle
+      (blocking_obstacle_start_s - adc_position_sl.s() > final_s_buffer)) {
+    // ADEBUG << "blocking obstacle distance: "
+    //        << blocking_obstacle_start_s - adc_position_sl->s();
+    return true;
+  } else {
+    return false;
+  }
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: `if (A) return true; else return false;` = `return A;`

File: modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc [hunk 6]
```diff
@@ -39,35 +41,122 @@ PathReuseDecider::PathReuseDecider(const TaskConfig& config)
...
+  auto* mutable_path_decider_status = PlanningContext::Instance()
+                                          ->mutable_planning_status()
+                                          ->mutable_path_decider();
+  // get blocking obstacle ID (front_static_obstacle_id)
+  const std::string& blocking_obstacle_ID =
+      mutable_path_decider_status->front_static_obstacle_id();
+  const IndexedList<std::string, Obstacle>& indexed_obstacles =
+      reference_line_info->path_decision()->obstacles();
+  const auto* blocking_obstacle = indexed_obstacles.Find(blocking_obstacle_ID);
+  if (blocking_obstacle == nullptr) return false;
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc]: Always add brackets for name scopes such as `if` and `for`.

**Pull request merged**